required/latex-lab/latex-lab-footnotes.dtx

```diff
@@ -0,0 +1,1731 @@
+%-------
+      \@footnotetext@cfgpointii       % config point
+        {
+%-------
+% ./resphilosophica/resphilosophica.cls
+%-------
+          \@footnotetext@cfgpointiii  % config point
+%-------
+% bibarts
+% fnbreak.sty
+          \UseHook{fntext/begin}
+%-------
+          \ignorespaces
+          #1
+%-------
+% bibarts
+% fnbreak.sty
+          \UseHook{fntext/end}
+%-------
+          \@footnotetext@cfgpointiv   % config point
+        }
+      \par
+    \color@endgroup
+  }
+%-------
+% ./linguex/linguex.sty
+  \UseHook{fntext/after}
+%-------
+}
+
+% default for config point (1 arg)
+\def\@footnotetext@cfgpoint    { \insert\footins }
+
+% default for config point (1 arg)
+\def\@footnotetext@cfgpointii  { \@makefntext }
+
+
+% default for config point (0 args)
+\def\@footnotetext@cfgpointiii { \rule\z@\footnotesep }
+
+% default for config point (0 args)
+\def\@footnotetext@cfgpointiv  { \@finalstrut\strutbox }
+
+
+% kernel hook for tagging (2 args)
+\cs_new:Npn \@kernel@process@footnotetext {}
+
+% Provide the name \LaTeXe{} is used to and do this unconditionally
+%   (no patching of class code if any). This means that if a class provides it own
+%    definition that gets lost and if necessary needs to be handled
+%    with firstaid (or updating of the class.
+
+\AddToHook{begindocument}{
+  \cs_set_eq:NN \@footnotetext \fnote_footnotetext:n
+}
+
+% alterations not covered:
+%
+% ./revtex4-1/revtex4-1.cls  ./revtex/ltxutil.sty ./revtex/revtex4-2.cls ... (need analysis)
+% ./bigfoot/bigfoot.sty
+
+
+
+% \footnotemargin is the logic implemented by footmisc. Perhaps we
+% don't want to do this like that in the kernel but for now I have
+% used this interface unchanged.
+
+
+\newdimen\footnotemargin
+\footnotemargin\maxdimen         % no value given
+
+\AtBeginDocument{
+  \ifdim \footnotemargin=\maxdimen
+    \setlength\footnotemargin{1.8em}
+  \fi
+}     
+
+
+\cs_new:Npn \fnote_makefntext:n #1 {
+  \UseHook{fntext/para2}
+  \@makefntext@cfgpoint
+      {
+        \ifdim\footnotemargin>\z@
+          \hb@xt@ \footnotemargin{\hss\@makefnmark}
+        \else
+          \ifdim\footnotemargin=\z@
+            \llap{\@makefnmark}
+          \else
+          \ifdim\footnotemargin=-\maxdimen
+               \@makefnmark
+            \else
+               \llap{\hb@xt@ -\footnotemargin{\@makefnmark\hss}}
+            \fi
+          \fi
+        \fi
+      }
+  \@makefntext@cfgpointii
+      { #1 }
+}
+
+
+
+
+% default for config point (1 arg)
+\def \@makefntext@cfgpoint { \noindent }
+
+% default for config point (1 arg)
+\long \def \@makefntext@cfgpointii #1 { #1 }
+
+
+
+%    If the definition for \cs{@makefntext} is that of the standard
+%    classes then replace it with \cs{fnote_makefntext:n}, otherwise
+%    try to patch the definition
+
+\newcommand\old@std@class@makefntext[1]{%
+    \parindent 1em%
+    \noindent
+    \hb@xt@1.8em{\hss\@makefnmark}#1}
+
+
+\tl_new:N \l__fnote_patch_tl
+\cs_new_protected:Npn \_fnote_patch:
```

Comment:
Could we have some comments here to explain what's going on?

Reply:
if I would understand it? I'm I clever enough to describe Bruno's code? I'll try ...

Reply:
Btw, `\_fnote` should be `\__fnote` (two underscores).  Basically my code looks for `\hbox` and, if found, it grabs `\hbox#1#{`.  Otherwise it looks for `\hb@xt@` and if found it grabs it and places the tagging thing there.  It is a bit unsafe for instance if `\hbox` appears without any brace group afterwards for some reason.

Reply:
made stuff internal in various places and use `@@` notation